accumulo-iterators/src/main/java/org/vertexium/accumulo/iterator/RowDeduplicationIterator.java

```diff
@@ -0,0 +1,218 @@
+package org.vertexium.accumulo.iterator;
+
+import org.apache.accumulo.core.conf.AccumuloConfiguration;
+import org.apache.accumulo.core.data.*;
+import org.apache.accumulo.core.iterators.IteratorEnvironment;
+import org.apache.accumulo.core.iterators.OptionDescriber;
+import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
+import org.apache.commons.collections.BufferOverflowException;
+import org.apache.hadoop.io.Text;
+
+import java.io.IOException;
+import java.util.*;
+
+public class RowDeduplicationIterator
+        implements SortedKeyValueIterator<Key, Value>, OptionDescriber {
+
+    public static final String MAX_BUFFER_SIZE_OPT = "maxBufferSize";
+    private static final long DEFAULT_MAX_BUFFER_SIZE = Long.MAX_VALUE;
+    private static final List<String> COLUMN_FAMILY_TYPES = Arrays.asList("PROP", "PROPD");
+    protected SortedKeyValueIterator<Key, Value> sourceIter;
+    private Key topKey = null;
+    private Value topValue = null;
+    private long maxBufferSize = DEFAULT_MAX_BUFFER_SIZE;
+
+    @Override
+    public SortedKeyValueIterator<Key, Value> deepCopy(IteratorEnvironment env) {
+        RowDeduplicationIterator newInstance;
```

Comment:
probably should just `new RowDeduplicationIterator()` here. Using reflection here probably not needed.